src/config.ts

```diff
@@ -1,13 +1,68 @@
-const config = {
-    REPL_FILE_EXT: 'calva-repl',
-    KEYBINDINGS_ENABLED_CONFIG_KEY: 'calva.keybindingsEnabled',
-    KEYBINDINGS_ENABLED_CONTEXT_KEY: 'calva:keybindingsEnabled'
-};
+import * as vscode from 'vscode';
+import { customREPLCommandSnippet } from './evaluate';
+import { ReplConnectSequence } from './nrepl/connectSequence';
+import { PrettyPrintingOptions } from './printer';
+
+const REPL_FILE_EXT = 'calva-repl';
+const KEYBINDINGS_ENABLED_CONFIG_KEY = 'calva.keybindingsEnabled';
+const KEYBINDINGS_ENABLED_CONTEXT_KEY = 'calva:keybindingsEnabled';
 
 type ReplSessionType = 'clj' | 'cljs';
 
-export {
-    ReplSessionType
+// include the 'file' and 'untitled' to the
+// document selector. All other schemes are
+// not known and therefore not supported.
+const documentSelector = [
+    { scheme: 'file', language: 'clojure' },
+    { scheme: 'jar', language: 'clojure' },
+    { scheme: 'untitled', language: 'clojure' }
+];
+
+/**
+ * Trims EDN alias and profile names from any surrounding whitespace or `:` characters.
+ * This in order to free the user from having to figure out how the name should be entered.
+ * @param  {string} name
+ * @return {string} The trimmed name
+ */
+ function _trimAliasName(name: string): string {
+    return name.replace(/^[\s,:]*/, "").replace(/[\s,:]*$/, "")
 }
 
-export default config;
+// TODO find a way to validate the configs
+function getWorkspaceConfig() {
```

Comment:
This should probably just be called `getConfig()`, as it is not just workspace configs in here.

Reply:
It's actually only workspace config that we're dealing with here, that's why I named it that way.

```javascript
const configOptions = vscode.workspace.getConfiguration('calva');
```

Reply:
But if you plan to maybe use this for other config in the future, then it makes sense to change it.

Reply:
Maybe I misunderstand here though. I was just going off `vscode.workspace.getConfiguration` and the fact that we only pull things from that thereafter, though I see `globalValue` at one point, so I'm unsure.

Reply:
For custom command snippets we merge the user config and the workspace config, since you can have commands that you want globally and commands that only make sense in a certain project. Therefore it is not just workspace config, as I see it. And also for the other settings, if there is no workspace config for a setting, the user config will be used. I think this is just config.

Reply:
Got it! Thanks. Will change.